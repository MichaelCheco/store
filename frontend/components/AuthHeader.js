import styled from 'styled-components';
import React from 'react';
import Link from 'next/link';
import User from './User'
const Name = styled.div`
  width: 25%;
  display: flex;
  color: white;
  font-size: 18px;
  margin-left: 50px;
  margin-top: 4px;
`;
const Div = styled.div`
  display: flex;
  background: black;
  justify-content: space-between;
  height: 45px;
`;
const Wrapper = styled.div`
  display: flex;
  align-items: center;
  width: 30%;
  a {
    margin: 0 10px;
    width: 25%;
    color: whitesmoke;
    font-weight: bold;
    font-size: 14px;
    white-space: nowrap;
    text-decoration: none;
    &:hover {
      text-decoration: underline;
    }
  }
`;

const AuthHeader = props => (
  <Div>
    <User>
      {({data: { me }}) => (
        me ? <Name>WELCOME {me.name.toUpperCase()}</Name> : <h5>.</h5>
        // <Name>Welcome {me.name}</Name>
      )}
    </User>
    <Wrapper>
      <Link href="/signup">
    <a>LOG IN</a>   
      </Link>
      <Link href="/signup"> 
    <a>CREATE AN ACCOUNT</a>
      </Link>
    </Wrapper>
  </Div>
)

export default AuthHeader;