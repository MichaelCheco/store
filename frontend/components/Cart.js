import React from 'react';
import { Query, Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import { adopt } from 'react-adopt';
import User from './User';
import calcTotalPrice from '../lib/calcTotalPrice';
import formatMoney from '../lib/formatMoney';
import CartStyles from './styles/CartStyles';
import Supreme from './styles/Supreme';
import CloseButton from './styles/CloseButton';
import Button from './styles/Button';
import CartItem from './CartItem';

/* client directive tells apollo to not go to the server 
for the data and instead to grab it from the apollo store */
const LOCAL_STATE_QUERY = gql`
	query {
		cartOpen @client
	}
`;

const TOGGLE_CART_MUTATION = gql`
	mutation {
		toggleCart @client
	}
`;

const Composed = adopt({
	user: ({ render }) => <User>{render}</User>,
	toggleCart: ({ render }) => <Mutation mutation={TOGGLE_CART_MUTATION}>{render}</Mutation>,
	localState: ({ render }) => <Query query={LOCAL_STATE_QUERY}>{render}</Query>,
});

const Cart = () => (
	<Composed>
		{({ user, toggleCart, localState }) => {
			const me = user.data.me;
			if (!me) return null;
			console.log(me);
			return (
				<CartStyles open={localState.data.cartOpen}>
					<header>
						<CloseButton onClick={toggleCart} title="close">
							&times;
						</CloseButton>
						<Supreme>{me.name}'s Cart</Supreme>
						<p>
							You Have {me.cart.length} Item{me.cart.length === 1 ? '' : 's'} in your cart
						</p>
					</header>
					<ul>
						{me.cart.map(cartItem => (
							<CartItem key={cartItem.id} cartItem={cartItem} />
						))}
					</ul>
					<footer>
						<p>{formatMoney(calcTotalPrice(me.cart))}</p>
						<Button>Checkout</Button>
					</footer>
				</CartStyles>
			);
		}}
	</Composed>
);
export default Cart;
export { LOCAL_STATE_QUERY, TOGGLE_CART_MUTATION };
