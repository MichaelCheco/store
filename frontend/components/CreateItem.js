import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import Router from 'next/router';
import Error from './ErrorMessage'

const CREATE_ITEM_MUTATION = gql`
 mutation CREATE_ITEM_MUTATION(
   $title: String!
   $description: String!
   $price: Int!
   $image: String
   $largeImage: String
   ) {
     createItem(
       title: $title
       description: $description
       price: $price
       image: $image
       largeImage: $largeImage
     ) {
       id
     }
   }
`;

class createItem extends Component {
  state = {
    title: '',
    description: '',
    image: '',
    largeImage: '',
    price: 1000,
  }
  handleChange = e => {
    const { type, name, value } = e.target;
    const val = type === 'number' ? parseFloat(value): value;
    this.setState({ [name]: val })
  }
  uploadFile = async e => {
    const files = e.target.files;
    const data = new FormData();
    data.append('file', files[0]);
    data.append('upload_preset', 'sickfits');
    const res = await fetch('https://api.cloudinary.com/v1_1/dpjplili1/image/upload',
    {
        method: 'POST',
        body: data
    })
    const file = await res.json()
    console.log(file, 'FILE')
    this.setState({
        image: file.secure_url,
        largeImage: file.eager[0].secure_url
    })  }
  render() {
    return (
      <Mutation mutation={CREATE_ITEM_MUTATION} variables={this.state}>
      {(createItem, { loading, error }) => (
        <form 
          onSubmit={async e => {
            e.preventDefault();

            const res = await createItem();
            console.log(res)
            Router.push({
              pathname: '/item',
              query: { id: res.data.createItem.id },
            });
          }}>
          <Error error={error} />
          <fieldset disabled={loading} aria-busy={loading}>
            <label htmlFor="file">
              Image
              <input
              type="file"
              id="file"
              name="file"
              placeholder="upload an image"
              required
              onChange={this.uploadFile}
              />
              {this.state.image && (
                <img width="200" src={this.state.image} alt="Upload Preview" />
               )}
            </label>
            <label htmlFor="title">
                Title
                <input
                type="text"
                id="title"
                name="title"
                placeholder="Title"
                required
                value={this.state.title}
                onChange={this.handleChange}
                />
            </label>
            <label htmlFor="price">
                Price
                <input
                type="text"
                id="price"
                name="price"
                placeholder="price"
                required
                value={this.state.price}
                onChange={this.handleChange}
                />
            </label>
            <label htmlFor="description">
                Description
                <input
                type="text"
                id="description"
                name="description"
                placeholder="description"
                required
                value={this.state.description}
                onChange={this.handleChange}
                />
            </label>
            <button type="submit">Submit</button>
          </fieldset>
          
          </form>
      )}
      </Mutation>
    )
  }
}

export default createItem;
export { CREATE_ITEM_MUTATION }