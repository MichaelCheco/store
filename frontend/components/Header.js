/* eslint-disable react/jsx-filename-extension */

import Link from 'next/link';
import Router from 'next/router';
import NProgress from 'nprogress';
import Wrapper from './styles/NavStyles';
import User from './User'
Router.onRouteChangeStart = () => {
  NProgress.start();
};
Router.onRouteChangeComplete = () => {
  NProgress.done();
};
Router.onRouteChangeError = () => {
  NProgress.done();
};


const Header = props => (
      <User>
        {({data: { me }}) => (
    <Wrapper>
      <Link href="/shop">
      <a>Shop 🛍️ </a>
      </Link>
      {me && (
        <>
      <Link href="/sell"><
        a>Sell 🤑</a>
        </Link>
      <Link href="/">
      <a>Account 👽</a>
      </Link>
      <Link href="/update">
      <a>Orders 📙</a>
      </Link>
      </>
      )}
        {!me && (
          <Link href="/signup">
      <a>Sign In 🤷‍♂️</a>
      </Link>
    )}
</Wrapper>
        )}
      </User>
);

export default Header;
