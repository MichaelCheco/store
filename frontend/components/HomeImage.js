import React from 'react';
import styled, { keyframes } from 'styled-components';

const Img = styled.img`
	width: 100%;
	object-fit: cover;
	height: 750px;
	opacity: 1;
	@media (max-width: 500px) {
	}
`;

const Wrapper = styled.div`
	border: 2px solid gray;
	position: absolute;
	background: whitesmoke;
	top: 28%;
	left: 25%;
	width: 55%;
	opacity: 0.8;
	height: auto;
	transform: skew(-4deg);
	padding: 1%;
	z-index: 2;
	h1 {
		text-align: center;
		font-size: 5.4rem;
	}
	p {
		font-weight: 600;
		line-height: 2;
		font-size: 2rem;
	}
	@media (max-width: 1200px) {
		h1 {
			font-size: 4.3rem;
		}
	}
	@media (max-width: 1100px) {
		h1 {
			font-size: 4.1rem;
		}
	}
	@media (max-width: 1000px) {
		h1 {
			font-size: 4rem;
		}
	}
	@media (max-width: 900px) {
		h1 {
			font-size: 3.7rem;
		}
	}
	@media (max-width: 850px) {
		h1 {
			font-size: 3.5rem;
		}
	}

	@media (max-width: 800px) {
		p {
			font-size: 1.8rem;
		}
		h1 {
			font-size: 3rem;
		}
	}
	@media (max-width: 680px) {
		h1 {
			font-size: 2.7rem;
		}
	}
	@media (max-width: 600px) {
		h1 {
			font-size: 2.2rem;
		}
		p {
			font-size: 1.7rem;
		}
	}
	@media (max-width: 500px) {
		width: 80%;
		height: 350px;
		top: 88%;
		left: 9%;
		transform: none;
		h1 {
			font-size: 2.5rem;
		}
		p {
			font-size: 1.6rem;
		}
	}
`;
const Image = () => (
	<>
		<Wrapper>
			<h1>Your Neighborhood Cafe</h1>
			<p>
				Born from the desire to deliver unequaled quality, Checo's Coffee was
				started with one thing and one thing in mind: Presenting nothing but the
				finest ingredients, processes, and technology to our customers. There
				are more cafes around than we care to count, but none do what we do.
			</p>
		</Wrapper>
		<Img src="static/ad.jpg" />
	</>
);

export default Image;
