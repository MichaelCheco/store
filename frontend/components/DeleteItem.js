import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import { ITEMS_QUERY } from './Items';
import styled from 'styled-components';
const Button = styled.div`
	border: 1px solid lightgray;
	padding: 10px;
	text-align: center;
	display: inline;
	margin-left: 5px;
	border-radius: 4px;
	cursor: pointer;
`;
const DELETE_ITEM_MUTATION = gql`
	mutation DELETE_ITEM_MUTATION($id: ID!) {
		deleteItem(id: $id) {
			id
		}
	}
`;

class DeleteItem extends Component {
	update = (cache, payload) => {
		const data = cache.readQuery({ query: ITEMS_QUERY });
		data.items = data.items.filter(
			item => item.id !== payload.data.deleteItem.id
		);
		cache.writeQuery({ query: ITEMS_QUERY, data });
	};
	render() {
		return (
			<Mutation
				mutation={DELETE_ITEM_MUTATION}
				variables={{ id: this.props.id }}
				update={this.update}>
				{(deleteItem, { error }) => (
					<Button
						onClick={() => {
							console.log(this.props.children);
							if (confirm('Are you sure you want to delete this item?')) {
								deleteItem().catch(err => {
									alert(err.message);
								});
							}
						}}>
						{this.props.children}
					</Button>
				)}
			</Mutation>
		);
	}
}

export default DeleteItem;
