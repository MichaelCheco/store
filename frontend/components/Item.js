import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Link from 'next/link'
import DeleteItem from './DeleteItem'
import styled from 'styled-components';

const Inner = styled.div`
    max-width: 400px;
    margin: 0 auto;
    padding: 2rem;
    border: 3px solid blue;
`;
const Img = styled.img`
width: 100%;
    height: 400px;
    object-fit: cover;
  
`;

export default class Item extends Component {
  render() {
    const { item } = this.props
    return (
      <Inner>
      <Link
        href={{
          pathname: '/item',
          query: { id: item.id },
        }}
        >
          <a>{item.title}</a>
        </Link>
        <p>{item.price}</p>
        <Img src={item.image} />
        <p>{item.description}</p>
        <Link
          href={{
            pathname: 'update',
            query: { id: item.id },
          }}>
            <a>Edit</a>
          </Link>
          <button>add to cart</button>
          <DeleteItem id={item.id}>Delete</DeleteItem>
      </Inner>
    )
  }
}