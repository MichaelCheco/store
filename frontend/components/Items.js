import React, { Component } from 'react';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import ErrorMessage from './ErrorMessage';
import Search from './Search';
import Item from './Item';
import Pagination from './Pagination';
import { perPage } from '../config';
import styled from 'styled-components';
import HomeImage from './HomeImage';
const Div = styled.div`
	border: 1px solid black;
`;
const ItemWrapper = styled.div`
	max-width: ${props => props.theme.maxWidth};
	margin: 0 auto;
	padding: 2rem;
	display: grid;
	grid-template-columns: 1fr 1fr 1fr 1fr;
	grid-template-rows: 60px 1fr;
	grid-gap: 10px;
	z-index: 1;
	background: white;
`;

const ITEMS_QUERY = gql`
 query ITEMS_QUERY($skip: Int = 0, $first: Int = ${perPage}) {
   items(skip: $skip, first: $first, orderBy: createdAt_DESC) {
     id
     title
     description
     price
     image
     largeImage
   }
 }
`;

export default class Items extends Component {
	render() {
		return (
			<Div>
				<ItemWrapper>
					<Search />
					<Pagination page={this.props.page} />
					<Query
						query={ITEMS_QUERY}
						variables={{
							skip: this.props.page * perPage - perPage,
						}}
					>
						{({ data, loading, error }) => {
							console.log(data, 'DATA');
							if (loading) return <h3>Loading</h3>;
							if (error) return <ErrorMessage error={error} />;
							return data.items.map(item => <Item item={item} key={item.id} />);
						}}
					</Query>
					{/* <Pagination page={this.props.page}/> */}
				</ItemWrapper>
			</Div>
		);
	}
}
export { ITEMS_QUERY };
