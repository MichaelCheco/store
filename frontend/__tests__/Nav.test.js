import { mount } from 'enzyme';
import wait from 'waait';
import toJSON from 'enzyme-to-json';
import Header from '../components/Header';
import { CURRENT_USER_QUERY } from '../components/User';
import { MockedProvider } from 'react-apollo/test-utils';
import { fakeUser, fakeCartItem } from '../lib/testUtils';

const notSignedInMocks = [
	{
		request: { query: CURRENT_USER_QUERY },
		result: { data: { me: null } },
	},
];
const signedInMocks = [
	{
		request: { query: CURRENT_USER_QUERY },
		result: { data: { me: fakeUser() } },
	},
];
const signedInMocksWithCartItems = [
	{
		request: { query: CURRENT_USER_QUERY },
		result: {
			data: {
				me: {
					...fakeUser(),
					cart: [fakeCartItem()],
				},
			},
		},
	},
];
describe('<Header/>', () => {
	it('renders a minimal nav when no user is signed in', async () => {
		const wrapper = mount(
			<MockedProvider mocks={notSignedInMocks}>
				<Header />
			</MockedProvider>
		);
		await wait();
		wrapper.update();

		const nav = wrapper.find('ul[data-test="nav"]');
		expect(toJSON(nav)).toMatchSnapshot();
	});

	xit('renders full nav when a user is signed in', async () => {
		const wrapper = mount(
			<MockedProvider mocks={signedInMocks}>
				<Header />
			</MockedProvider>
		);
		await wait();
		wrapper.update();
		const nav = wrapper.find('ul[data-test="nav"]');
		expect(nav.children().length()).toBe(5);
		// expect(toJSON(nav)).toMatchSnapshot();
	});

	xit('renders the amount of items in the cart', async () => {
		const wrapper = mount(
			<MockedProvider mocks={signedInMocksWithCartItems}>
				<Header />
			</MockedProvider>
		);
		await wait();
		wrapper.update();

		const nav = wrapper.find('ul[data-test="nav"]');
		const count = nav.find('div.count');
		console.log(count.debug());
		expect(toJSON(count)).toMatchSnapshot();
	});
});
