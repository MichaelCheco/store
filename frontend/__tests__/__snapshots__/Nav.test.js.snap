// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> renders a minimal nav when no user is signed in 1`] = `
<ul
  className="NavStyles-x3z0n-0 gyNVsA"
  data-test="nav"
>
  <Link
    href="/shop"
  >
    <a
      href="/shop"
      onClick={[Function]}
    >
      Shop
    </a>
  </Link>
</ul>
`;

exports[`<Header/> renders the amount of items in the cart 1`] = `null`;
